src/main/java/com/jfrog/ide/idea/navigation/NavigationService.java

```diff
@@ -48,15 +48,32 @@ public void addNavigation(DependenciesTree treeNode, PsiElement navigationTarget
         if (document == null) {
             return;
         }
-
         NavigationTarget navigationTarget = new NavigationTarget(navigationTargetElement, document.getLineNumber(navigationTargetElement.getTextOffset()));
         Set<NavigationTarget> navigationTargets = navigationMap.get(treeNode);
         if (navigationTargets == null) {
             navigationTargets = new HashSet<>(Collections.singletonList(navigationTarget));
             navigationMap.put(treeNode, navigationTargets);
             return;
         }
-        navigationTargets.add(navigationTarget);
+        if (!isRedundantNavigation(navigationTargets, navigationTarget)) {
+            navigationTargets.add(navigationTarget);
+        }
+    }
+
+    /**
+     * A navigation is redundant when a navigation to the same file and line-number already exists.
+     * @param existingNavigations - Set of existing registered navigations from a node in issues-tree.
+     * @param newNavigation - The new navigation candidate for addition.
+     * @return true if the new navigation candidate is redundant.
+     */
+    private boolean isRedundantNavigation(Set<NavigationTarget> existingNavigations, NavigationTarget newNavigation) {
+        for (NavigationTarget existingTarget : existingNavigations) {
+            if (existingTarget.getLineNumber() == newNavigation.getLineNumber() &&
+                    existingTarget.getElement().getContainingFile().equals(newNavigation.getElement().getContainingFile())) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Instead of doing all of this, can we just change the `equals` and `hashCode` of `NavigationTarget` to compare the containing file of the PSI? I mean:
```java
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof NavigationTarget)) return false;
        NavigationTarget that = (NavigationTarget) o;
        return lineNumber == that.lineNumber &&
                Objects.equals(element.getContainingFile(), that.element.getContainingFile());
    }

    @Override
    public int hashCode() {
        return Objects.hash(element.getContainingFile(), lineNumber);
    }
```

Reply:
@yahavi in the beginning that's exactly what I did, but then I thought that it might be confusing in other (future) usages, as such behaviour isn't what you usually expect.
WDYT?

Reply:
I think the main reason to use set is to avoid duplication. The second reason is to insert and remove in O(1) using hash set.
For me, if you are using set you want to satisfy the both requirements above.
We miss both when we are using set with duplications and using a `for` loop to iterate in O(n).

Reply:
Ok, let's do this.